lua/nvim-treesitter/configs.lua

```diff
@@ -1,88 +1,327 @@
+      return queries.get_query(ft, 'highlights') ~= nil
+    end
+  },
+  -- selection = {
+  --   enable = false,
+  --   disable = {},
+  --   keymaps = {},
+  --   is_enabled = function() return false end
+  -- },
+  -- folding = {
+  --   enable = false,
+  --   disable = {},
+  --   keymaps = {},
+  --   is_enabled = function() return false end
+  -- }
+}
+
+local M = {}
+
+local function enable_module(mod, bufnr, ft)
+  local bufnr = bufnr or api.nvim_get_current_buf()
+  local ft = ft or api.nvim_buf_get_option(bufnr, 'ft')
+  if not parsers[ft] or not config[mod] then
+    return
+  end
+
+  local str = "lua require'nvim-treesitter.%s'.attach(%s, %s)"
+  local cmd = string.format(str, mod, bufnr, ft)
+  api.nvim_command(cmd)
+end
+
+local function disable_module(mod, bufnr, ft)
+  local bufnr = bufnr or api.nvim_get_current_buf()
+  local ft = ft or api.nvim_buf_get_option(bufnr, 'ft')
+  if not parsers[ft] or not config[mod] then
+    return
+  end
+
+  local str = "lua require'nvim-treesitter.%s'.detach(%s)"
+  local cmd = string.format(str, mod, bufnr, ft)
+  api.nvim_command(cmd)
+end
+
+local function enable_all(mod, ft)
+  for _, bufnr in pairs(api.nvim_list_bufs()) do
+    if not ft or api.nvim_buf_get_option(bufnr, 'ft') == ft then
+      enable_module(mod, bufnr, ft)
+    end
+  end
+end
+
+local function disable_all(mod, ft)
+  for _, bufnr in pairs(api.nvim_list_bufs()) do
+    if not ft or api.nvim_buf_get_option(bufnr, 'ft') == ft then
+      disable_module(mod, bufnr, ft)
+    end
+  end
+end
+
+M.commands = {
+  TSEnable = {
+    run = enable_module,
+    args = {
+      "-nargs=1",
+      "-complete=custom,v:lua.ts_available_modules"
+    },
+    description = '`:TSEnable module_name` enable a specified module on the current buffer'
+  },
+  TSDisable = {
+    run = disable_module,
+    args = {
+      "-nargs=1",
+      "-complete=custom,v:lua.ts_available_modules"
+    },
+    description = '`:TSDisable module_name` disable a specified module on the current buffer'
+  },
+  TSEnableAll = {
+    run = enable_all,
+    args = {
+      "-nargs=+",
+      "-complete=custom,v:lua.ts_available_modules"
+    },
+    description = '`:TSEnableAll module_name (filetype)` enables a specified module on all buffers. If filetype is specified, enable only for specified filetype'
+  },
+  TSDisableAll = {
+    run = disable_all,
+    args = {
+      "-nargs=+",
+      "-complete=custom,v:lua.ts_available_modules"
+    },
+    description = '`:TSDisableAll module_name (filetype)` disables a specified module on all buffers. If filetype is specified, disable only for specified filetype'
+  },
+}
+
+-- @param mod: module (string)
+-- @param ft: filetype (string)
+function M.is_enabled(mod, ft)
+  if not M.get_parser_configs()[ft] then return false end
+
+  local module_config = M.get_config()[mod]
+  if not module_config then return false end
+
+  if module_config.enable == false or not module_config.is_enabled(ft) then
```

Comment:
I'm a bit rought but maybe change it to this :
```suggestion
  if not module_config.enable or not module_config.is_enabled(ft) then
```